### PR TITLE
GLTFLoader: Clean up buildNodeHierarchy

### LINKED ISSUE #15561: GLTFExporter reorders children in groups.
##### Description of the problem

When exporting a model via GLTFExporter, it reorders children in groups (it places groups first and than meshes). For example if my object children structure is [mesh, mesh, group, mesh], the exported object structure will be [group, mesh, mesh, mesh].

Model before export ([download here](https://we.tl/t-zQdXd1S1ly)) has a group somewhere in hierarchy
2019-01-11 14_38_04-window

After the export group is placed in the beginning of hierarchy
2019-01-11 14_38_25-window

This behavior is unwanted, as I'm assigning materials based on child index. Would it be possible to adjust the GLTFExporter so it keeps the original hierarchy, in the way that importing and exporting the same model would result in the exact same data?

##### Three.js version
- [x] 98

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

### PR DESCRIPTION
**Pull request by takahirox**
This PR cleans up `buildNodeHierarchy()` in `GLTFLoader`.

Changes
- Move node hierarchy building and `Skeleton` initialization to `.loadNode()` from `.loadScene()`. With this change 
  - `.getDependency('node', nodeIndex)` will return a node (Three.js object) with children underneath.
  - `.loadScene()` will be much simpler
- Make Three.js object children in the same order as gltf `nodeDef.children`. Fixes #15561

### CODE REVIEW

File: examples/js/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -3182,7 +3182,68 @@ THREE.GLTFLoader = ( function () {
...
+										mat.fromArray( skinEntry.inverseBindMatrices.array, j * 16 );
+
+									}
+
+									boneInverses.push( mat );
+
+								} else {
+
+									console.warn( 'THREE.GLTFLoader: Joint "%s" could not be found.', skinEntry.joints[ j ] );
+
```
Comment (donmccurdy) [inline examples/js/loaders/GLTFLoader.js]: Any idea why we have the outer IIFE (starting with the `return ( function() {...` on line 3132? This is nested deeply enough that it's hard to read for me; removing the IIFE might help, or perhaps moving some of the skin-related code into a helper function here.

**Comment by takahirox** [comment]
Good point. Let me try to think how to resolve...

File: examples/js/loaders/GLTFLoader.js (hunk 1 above)
Comment (takahirox) [reply examples/js/loaders/GLTFLoader.js]: Seems like from #15350

**Comment by Mugen87** [comment]
Any updates on this PR? It seems the concerns of @donmccurdy are not yet addressed, right?

**Comment by takahirox** [comment]
Yes, you're right. I've been a bit too busy so give me time. Or does this PR block any other PRs?

**Comment by Mugen87** [comment]
Not that I'm aware of. I just wanted to check the status of this PR. Sometimes, issues are already solved with different ones...

**Comment by mrdoob** [comment]
Closing this for now.

**Pull request closed without merging**